docs/content/docs/dev/table/sql/queries/hints.md

```diff
@@ -84,4 +84,323 @@ insert into kafka_table1 /*+ OPTIONS('sink.partitioner'='round-robin') */ select
+Users can configure the async parameters via async options on async lookup mode.
+
+Example:
+```sql
+-- configure the async parameters: 'output-mode', 'capacity', 'timeout', can set single one or multi params
+LOOKUP('table'='Customers', 'async'='true', 'output-mode'='allow_unordered', 'capacity'='100', 'timeout'='180s')
+```
+Note: the async options are consistent with the async options in [job level Execution Options]({{< ref "docs/dev/table/config" >}}#execution-options),
+will use job level configuration if not set. Another difference is that the scope of the LOOKUP hint
+is smaller, limited to the table name corresponding to the hint option set in the current lookup
+operation (other lookup operations will not be affected by the LOOKUP hint).
+
+e.g., if the job level configuration is:
+```gitexclude
+table.exec.async-lookup.output-mode: ORDERED
+table.exec.async-lookup.buffer-capacity: 100
+table.exec.async-lookup.timeout: 180s
+```
+
+then the following hints:
+```sql
+1. LOOKUP('table'='Customers', 'async'='true', 'output-mode'='allow_unordered')
+2. LOOKUP('table'='Customers', 'async'='true', 'timeout'='300s')
+```
+
+are equivalent to:
+```sql
+1. LOOKUP('table'='Customers', 'async'='true', 'output-mode'='allow_unordered', 'capacity'='100', 'timeout'='180s')
+2. LOOKUP('table'='Customers', 'async'='true', 'output-mode'='ordered', 'capacity'='100', 'timeout'='300s')
+```
+
+##### 3. Enable Delayed Retry Strategy For Lookup
+Delayed retry for lookup join is intended to solve the problem of delayed updates in external system
+which cause unexpected enrichment with stream data. The hint option 'retry-predicate'='lookup_miss'
+can enable retry on both sync and async lookup, only fixed delay retry strategy is supported currently.
+
+Options of fixed delay retry strategy:
+```gitexclude
+'retry-strategy'='fixed_delay'
+-- fixed delay duration
+'fixed-delay'='10s'
+-- max number of retry(counting from the retry operation, if set to '1', then a single lookup process
+-- for a specific lookup key will actually execute up to 2 lookup requests)
+'max-attempts'='3'
+```
+
+Example:
+1. enable retry on async lookup
+```sql
+LOOKUP('table'='Customers', 'async'='true', 'retry-predicate'='lookup_miss', 'retry-strategy'='fixed_delay', 'fixed-delay'='10s','max-attempts'='3')
+```
+
+2. enable retry on sync lookup
+```sql
+LOOKUP('table'='Customers', 'async'='false', 'retry-predicate'='lookup_miss', 'retry-strategy'='fixed_delay', 'fixed-delay'='10s','max-attempts'='3')
+```
+
+If the lookup source only has one capability, then the 'async' mode option can be omitted:
+
+```sql
+LOOKUP('table'='Customers', 'retry-predicate'='lookup_miss', 'retry-strategy'='fixed_delay', 'fixed-delay'='10s','max-attempts'='3')
+```
+
+##### Further Notes
+
+###### Effect Of Enabling Caching On Retries
+[FLIP-221](https://cwiki.apache.org/confluence/display/FLINK/FLIP-229%3A+Introduces+Join+Hint+for+Flink+SQL+Batch+Job) adds caching support for lookup source,
```

Comment:
It seems that this url is wrong.

Reply:
+1

Reply:
The url should be: https://cwiki.apache.org/confluence/display/FLINK/FLIP-221%3A+Abstraction+for+lookup+source+cache+and+metric